### PR TITLE
feat: added config 'disableOnNilValue' to disable tippy if value is nil

### PR DESCRIPTION
**Pull request by florianjung79**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

When a value equal to null or undefined is assigned to the tippy directive an error is thrown as none of them is a supported type. Although an error is thrown, an empty tooltip will be instantiated. In order to disable the tooltip for nil values, one has to use the isEnable flag, which.

Issue Number: N/A

## What is the new behavior?

When the new global config `disableOnNilValue` is set to `true` and the value assigned to the tippy directive is `null` or `undefined`, the tippy tooltip won't be rendered.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

This feature can be seen as a shortcut to use the directive together with `isEnable`.

### CODE REVIEW

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts [hunk 1]
```diff
@@ -202,7 +202,16 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
         this.globalConfig.onCreate?.(instance);
       },
       onShow: instance => {
-        this.zone.run(() => this.instance.setContent(this.resolveContent()));
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/tippy.directive.ts:205]: 1. We should not create the instance if the value is null or undefined.
2. We should listen to changes and perform a re-check.

Comment (NetanelBasal) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:205]: I don't know if option two is so common. Maybe I'd just start with not creating the instance if [tippy] is nil

Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:205]: 1) The idea was to keep as much existing behaviour and use the documented behaviour of the onShow callback. Do you think it is more appropriate to guard the creation of the instance itself? I will make some modifications. `createInstance` is used three times. I think I am going to guard the method call... Do you have any thoughts on this?

2) I am not sure either, but I will take a look into it.

Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:205]: Addressed 1) in the last commit.

**Comment by florianjung79** [comment]
I'll agree with you that the re-check should be a use case that doesn't occurs this often. Let's postpone this change for as long as it is not needed.

**Comment by florianjung79** [comment]
Hi Netanel, is there anything I can do to have the suggested changes being reviewed?

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts [hunk 2]
```diff
@@ -131,7 +131,7 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
         inView(this.host)
           .pipe(takeUntil(this.destroyed))
           .subscribe(() => {
-            this.createInstance();
+            this.createInstanceOnNonNilValue();
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/tippy.directive.ts:134]: In your case, the value you pass to tippy is static or dynamic?

Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:134]: It may change depending on the domain model, so it's dynamic in some cases.

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts [hunk 3]
```diff
@@ -185,6 +185,12 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
     return `${this.host.nativeElement.getBoundingClientRect().width}px`;
   }
 
+  private createInstanceOnNonNilValue() {
+    if (this.globalConfig.disableOnNilValue && this.content !== null && this.content !== undefined) {
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/tippy.directive.ts:189]: Why do we need a config for it? It doesn't make sense anyway to show a tooltip when the value is nil. We can make it the default functionality.

Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:189]: > It doesn't make sense anyway to show a tooltip when the value is nil. We can make it the default functionality.

We absolutely can. I tried to keep the existing behaviour as is and add the new behaviour via global config. I agree that it doesn't make sense at all. That's why I tried to discuss the idea first. So if you agree, I will remove the global config part and stick to skipping the tooltip creation, if the content is nil.

Comment (NetanelBasal) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:189]: Sure

Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts]: I have removed the global config and reworked the docs accordingly.

File: cypress/integration/helipopper.spec.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ describe("@ngneat/helipopper", () => {
   describe("Custom template", () => {
     it("should create tooltip with a custom template", () => {
       cy.get("#custom-template button")
-        .click()
+        .click({ force: true })
```
Comment (NetanelBasal) [inline cypress/integration/helipopper.spec.ts:32]: Why did you change it?

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts [hunk 5]
```diff
@@ -185,6 +185,12 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
     return `${this.host.nativeElement.getBoundingClientRect().width}px`;
   }
 
+  private createInstanceOnNonNilValue() {
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/tippy.directive.ts:188]: There is no need for another method. Add it inside the createInstance()

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts [hunk 6]
```diff
@@ -225,6 +231,10 @@ export class TippyDirective implements OnChanges, AfterViewInit, OnDestroy, OnIn
   }
 
   private resolveContent() {
+    if (this.content === null || this.content === undefined) {
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/tippy.directive.ts:234]: How do we get it here if we don't create an instance?

File: cypress/integration/helipopper.spec.ts (hunk 4 above)
Comment (florianjung79) [reply cypress/integration/helipopper.spec.ts:32]: I changed it because the button is out of view and forcing the click skips the actionability check, which imho is not essential here.

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts (hunk 6 above)
Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts]: Sorry, you're right. I missed that. It will be removed with the next push.

File: projects/ngneat/helipopper/src/lib/tippy.directive.ts (hunk 5 above)
Comment (florianjung79) [reply projects/ngneat/helipopper/src/lib/tippy.directive.ts:188]: It will be added inside the createInstance() method with the next push.

**Review by NetanelBasal — APPROVED** [review]
